### PR TITLE
build: if draft, return early

### PR DESCRIPTION
**Pull request by umarcor**
# Description

This is a small change to make the source easier to read. The condition that evaluates `draft` can return early, because there is no content after the else branch. As a result, ~25 lines can be unindented.

# Checklist

* [x] Make sure changes are covered by existing or new tests.
* [x] For at least one Python version, make sure local test run is green.
* [x] Create a file in `src/towncrier/newsfragments/`. Describe your
  change and include important information. Your change will be included in the public release notes.
* [ ] Make sure all GitHub Actions checks are green (they are automatically checking all of the above).
* [x] Ensure `docs/tutorial.rst` is still up-to-date.
* [x] If you add new **CLI arguments** (or change the meaning of existing ones), make sure `docs/cli.rst` reflects those changes.
* [x] If you add new **configuration options** (or change the meaning of existing ones), make sure `docs/configuration.rst` reflects those changes.

### CODE REVIEW

**Review by hynek — CHANGES_REQUESTED** [review]
The changes look fine, but it needs a newsfragment.

Running `towncrier create -c "" 486.misc` should be fine.

Thanks!

**Comment by umarcor** [comment]
@hynek, done!

**Comment by hynek** [comment]
I've just noticed that #483 is touching the same lines, so I'm gonna hold this back, until it lands soon, because the conflicts should be much easier to fix here than there. (if it gets protracted, we can move this on forward).

**Review by hynek — APPROVED** [review]
OK looks like Chris is busy, let's get over with this.

**Pull request merged**
